### ISSUE TITLE
Maven Plugin: Generated files are always detected as "Stale" by java compiler.

### ISSUE THREAD

**Issue by jimmymain** [joelittlejohn/jsonschema2pojo#1322]
The files generated by the plugin should be checked with a hash code or other mechanism, and not re-generated if the contents of the file have not changed.

By creating / overwriting the generated files in the target directory, mvn compile always detects the files have changed.
The increases build time for no reason.

Is there a workaround / solution for this, or is it something that needs to be added to the plugin?

```
[DEBUG] Stale source detected: /Users/s/IdeaProjects/cs/target/generated-sources/com/ck/api/response/Application.java
[DEBUG] Stale source detected: /Users/s/IdeaProjects/cs/target/generated-sources/com/ck/api/response/Inflows.java
[DEBUG] Stale source detected: ...
```
and so on.

**Comment by joelittlejohn** [comment]
Hi Jim

> The files generated by the plugin should be checked with a hash code or other mechanism, and not re-generated if the contents of the file have not changed.

This plugin does not maintain old source code or access to source code generated previously, so there's no way to compare old with new.

> mvn compile always detects the files have changed

I believe it's the Maven compile plugin that should be skipping compilation if the source is identical (by hash) to the source used to generate the previously compiled file.

Are you experiencing this bug by any chance:

https://issues.apache.org/jira/browse/MCOMPILER-209

?

**Comment by jimmymain** [comment]
Hi @joelittlejohn,
Thanks very much for the quick response.

Perhaps you are right, and I agree with the last comment on the bug, maybe it's time for gradle.

I am however still confused as to how the utility (plugin) would generate files, and at the same time not be able to read what's being replaced and not replace if it's identical?

**Comment by tonketonky** [comment]
Hi @joelittlejohn,

I did a bit of an investigation and it really seems like a bug in `jsonschema2pojo-maven-plugin`.

I encountered the same issue and found out it works in version `1.0.0-alpha2` but does not work in version `1.0.0-alpha4` (there is a broken maven plugin`1.0.0-alpha3` so I have not tried).

This is the output of `mvn compile` when using `jsonschema2pojo-maven-plugin:1.0.0-alpha2` :
```
[INFO] 
[INFO] --- jsonschema2pojo:1.0.0-alpha2:generate (default) @ my-module ---
[INFO] 
[INFO] --- resources:3.3.0:resources (default-resources) @ my-module ---
[INFO] Copying 47 resources
[INFO] 
[INFO] --- compiler:3.11.0:compile (default-compile) @ my-module ---
[INFO] Nothing to compile - all classes are up to date
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.956 s
[INFO] Finished at: 2023-08-11T14:59:24+02:00
[INFO] ------------------------------------------------------------------------

```

And this is the output of `mvn compile` when using `jsonschema2pojo-maven-plugin:1.0.0-alpha4` :
```
[INFO] 
[INFO] --- jsonschema2pojo:1.0.0-alpha4:generate (default) @ my-module ---
[INFO] 
[INFO] --- resources:3.3.0:resources (default-resources) @ my-module ---
[INFO] Copying 47 resources
[INFO] 
[INFO] --- compiler:3.11.0:compile (default-compile) @ my-module ---
[INFO] Changes detected - recompiling the module! :source
[INFO] Compiling 90 source files with javac [debug target 17] to target/classes
...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.171 s
[INFO] Finished at: 2023-08-11T14:59:43+02:00
[INFO] ------------------------------------------------------------------------

```

I noticed that when using `jsonchema2pojo-maven-plugin:1.0.0-alpha4` files generated under `/target/classes/` have changed last modified date each time I run `mvn compile`. This was not happening with `jsonschema2pojo-maven-plugin:1.0.0-alpha2`. I suspect this to be the root cause that makes maven always recompile sources.

I am aware that this is an old and closed issue. Maybe shall I created a fresh new issue for that?

Thanks

**Comment by joelittlejohn** [comment]
Could you try adding the `-X` option to Maven to see what is the stale source file?

I don't see anything obvious between these releases to explain this yet. If you run the plugin, copy all sources to another dir, then run the plugin again, can you detect difference between the source files generated the first time and the second time?

**Comment by joelittlejohn** [comment]
It's not a trivial task, but if you get things set up nicely, you could bisect the log between alpha2 and alpha4 (`mvn install` each time) to narrow this down.

**Comment by tonketonky** [comment]
Thanks for the quick response. 

And sorry, I made a mistake describing my observations. When I wrote `/target/classes/` I meant `/target/generated-sources/` - files with changed last modified date are under `/target/generated-sources`.

I tried to reduce json schemas in the module to just 1 and ran it with `-X` option and here is the output:
```
...
[DEBUG] CompilerReuseStrategy: reuseCreated
[DEBUG] useIncrementalCompilation enabled
[DEBUG] Stale source detected: /path/to/generated/Class.java
[INFO] Changes detected - recompiling the module! :source
...

```
So there is just this one file detected as stale.

I also tried creating a diff between debug logs from `alpha2` and `alpha4` and here it is:
```
< [DEBUG] Dependency collection stats {ConflictMarker.analyzeTime=582730, ConflictMarker.markTime=134860, ConflictMarker.nodeCount=35, ConflictIdSorter.graphTime=434244, ConflictIdSorter.topsortTime=314761, ConflictIdSorter.conflictIdCount=19, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=3365288, ConflictResolver.conflictItemCount=34, DfDependencyCollector.collectTime=107135769, DfDependencyCollector.transformTime=6529839}
---
> [DEBUG] Dependency collection stats {ConflictMarker.analyzeTime=632985, ConflictMarker.markTime=169156, ConflictMarker.nodeCount=35, ConflictIdSorter.graphTime=427201, ConflictIdSorter.topsortTime=293091, ConflictIdSorter.conflictIdCount=19, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=3794057, ConflictResolver.conflictItemCount=34, DfDependencyCollector.collectTime=135411251, DfDependencyCollector.transformTime=6994187}
133c133
< [DEBUG] Dependency collection stats {ConflictMarker.analyzeTime=1973007, ConflictMarker.markTime=1027889, ConflictMarker.nodeCount=736, ConflictIdSorter.graphTime=1926133, ConflictIdSorter.topsortTime=204581, ConflictIdSorter.conflictIdCount=248, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=33305553, ConflictResolver.conflictItemCount=705, DfDependencyCollector.collectTime=559392559, DfDependencyCollector.transformTime=38536354}
---
> [DEBUG] Dependency collection stats {ConflictMarker.analyzeTime=1915134, ConflictMarker.markTime=956716, ConflictMarker.nodeCount=736, ConflictIdSorter.graphTime=1447912, ConflictIdSorter.topsortTime=207831, ConflictIdSorter.conflictIdCount=248, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=35355639, ConflictResolver.conflictItemCount=705, DfDependencyCollector.collectTime=589426823, DfDependencyCollector.transformTime=40022959}
669c669
< [DEBUG] Goal:          org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha2:generate (default)
---
> [DEBUG] Goal:          org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha4:generate (default)
701c701
<   <outputDirectory default-value="${project.build.directory}/java-gen">/home/tony/path/to/my/module/target/generated-sources</outputDirectory>
---
>   <outputDirectory default-value="${project.build.directory}/generated-sources/jsonschema2pojo">/home/tony/path/to/my/module/target/generated-sources</outputDirectory>
717c717`
<   <targetVersion default-value="1.6">${maven.compiler.target}</targetVersion>
---
>   <targetVersion default-value="${maven.compiler.target}">${jsonschema2pojo.targetJavaVersion}</targetVersion>
799c799
< [DEBUG] Dependency collection stats {ConflictMarker.analyzeTime=235611, ConflictMarker.markTime=98726, ConflictMarker.nodeCount=61, ConflictIdSorter.graphTime=91416, ConflictIdSorter.topsortTime=79062, ConflictIdSorter.conflictIdCount=32, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=1503664, ConflictResolver.conflictItemCount=60, DfDependencyCollector.collectTime=111400824, DfDependencyCollector.transformTime=2098194}
---
> [DEBUG] Dependency collection stats {ConflictMarker.analyzeTime=249678, ConflictMarker.markTime=62586, ConflictMarker.nodeCount=61, ConflictIdSorter.graphTime=84858, ConflictIdSorter.topsortTime=27408, ConflictIdSorter.conflictIdCount=32, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=1763854, ConflictResolver.conflictItemCount=60, DfDependencyCollector.collectTime=101029548, DfDependencyCollector.transformTime=2285863}
835,836c835
< [INFO] --- jsonschema2pojo:1.0.0-alpha2:generate (default) @ my-module ---
< [DEBUG] Using mirror maven-default-http-blocker (http://0.0.0.0/) for apache.snapshots (http://people.apache.org/repo/m2-snapshot-repository).
---
> [INFO] --- jsonschema2pojo:1.0.0-alpha4:generate (default) @ my-module ---
838,841c837,840
< [DEBUG] Dependency collection stats {ConflictMarker.analyzeTime=199554, ConflictMarker.markTime=92904, ConflictMarker.nodeCount=74, ConflictIdSorter.graphTime=40905, ConflictIdSorter.topsortTime=55586, ConflictIdSorter.conflictIdCount=50, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=1786030, ConflictResolver.conflictItemCount=74, DfDependencyCollector.collectTime=108508305, DfDependencyCollector.transformTime=2201955}
< [DEBUG] org.jsonschema2pojo:jsonschema2pojo-maven-plugin:jar:1.0.0-alpha2
< [DEBUG]    org.jsonschema2pojo:jsonschema2pojo-core:jar:1.0.0-alpha2:compile
< [DEBUG]       org.jsonschema2pojo:jsonschema2pojo-scalagen:jar:1.0.0-alpha2:compile
---
> [DEBUG] Dependency collection stats {ConflictMarker.analyzeTime=166016, ConflictMarker.markTime=45474, ConflictMarker.nodeCount=62, ConflictIdSorter.graphTime=24567, ConflictIdSorter.topsortTime=30401, ConflictIdSorter.conflictIdCount=39, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=1116583, ConflictResolver.conflictItemCount=62, DfDependencyCollector.collectTime=67074412, DfDependencyCollector.transformTime=1403782}
> [DEBUG] org.jsonschema2pojo:jsonschema2pojo-maven-plugin:jar:1.0.0-alpha4
> [DEBUG]    org.jsonschema2pojo:jsonschema2pojo-core:jar:1.0.0-alpha4:compile
> [DEBUG]       org.jsonschema2pojo:jsonschema2pojo-scalagen:jar:1.0.0-alpha4:compile
851,859d849
< [DEBUG]       com.google.android:android:jar:4.1.1.4:compile (version managed from default)
< [DEBUG]          commons-logging:commons-logging:jar:1.1.1:compile
< [DEBUG]          org.apache.httpcomponents:httpclient:jar:4.0.1:compile
< [DEBUG]             org.apache.httpcomponents:httpcore:jar:4.0.1:compile
< [DEBUG]             commons-codec:commons-codec:jar:1.9:test (scope managed from default) (version managed from default)
< [DEBUG]          org.khronos:opengl-api:jar:gl1.1-android-2.1_r1:compile
< [DEBUG]          xerces:xmlParserAPIs:jar:2.6.2:compile
< [DEBUG]          xpp3:xpp3:jar:1.1.4c:compile
< [DEBUG]          org.json:json:jar:20080701:compile
863,864d852
< [DEBUG]       com.squareup.moshi:moshi:jar:1.5.0:compile (version managed from default)
< [DEBUG]          com.squareup.okio:okio:jar:1.13.0:compile
889,890c877,878
< [DEBUG] Created new class realm plugin>org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha2
< [DEBUG] Importing foreign packages into class realm plugin>org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha2
---
> [DEBUG] Created new class realm plugin>org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha4
> [DEBUG] Importing foreign packages into class realm plugin>org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha4
892,895c880,883
< [DEBUG] Populating class realm plugin>org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha2
< [DEBUG]   Included: org.jsonschema2pojo:jsonschema2pojo-maven-plugin:jar:1.0.0-alpha2
< [DEBUG]   Included: org.jsonschema2pojo:jsonschema2pojo-core:jar:1.0.0-alpha2
< [DEBUG]   Included: org.jsonschema2pojo:jsonschema2pojo-scalagen:jar:1.0.0-alpha2
---
> [DEBUG] Populating class realm plugin>org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha4
> [DEBUG]   Included: org.jsonschema2pojo:jsonschema2pojo-maven-plugin:jar:1.0.0-alpha4
> [DEBUG]   Included: org.jsonschema2pojo:jsonschema2pojo-core:jar:1.0.0-alpha4
> [DEBUG]   Included: org.jsonschema2pojo:jsonschema2pojo-scalagen:jar:1.0.0-alpha4
905,912d892
< [DEBUG]   Included: com.google.android:android:jar:4.1.1.4
< [DEBUG]   Included: commons-logging:commons-logging:jar:1.1.1
< [DEBUG]   Included: org.apache.httpcomponents:httpclient:jar:4.0.1
< [DEBUG]   Included: org.apache.httpcomponents:httpcore:jar:4.0.1
< [DEBUG]   Included: org.khronos:opengl-api:jar:gl1.1-android-2.1_r1
< [DEBUG]   Included: xerces:xmlParserAPIs:jar:2.6.2
< [DEBUG]   Included: xpp3:xpp3:jar:1.1.4c
< [DEBUG]   Included: org.json:json:jar:20080701
916,917d895
< [DEBUG]   Included: com.squareup.moshi:moshi:jar:1.5.0
< [DEBUG]   Included: com.squareup.okio:okio:jar:1.13.0
929,930c907,908
< [DEBUG] Loading mojo org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha2:generate from plugin realm ClassRealm[plugin>org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha2, parent: jdk.internal.loader.ClassLoaders$AppClassLoader@5cb0d902]
< [DEBUG] Configuring mojo execution 'org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha2:generate:default' with basic configurator -->
---
> [DEBUG] Loading mojo org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha4:generate from plugin realm ClassRealm[plugin>org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha4, parent: jdk.internal.loader.ClassLoaders$AppClassLoader@5cb0d902]
> [DEBUG] Configuring mojo execution 'org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha4:generate:default' with basic configurator -->
1006c984
< [DEBUG] Dependency collection stats {ConflictMarker.analyzeTime=41417, ConflictMarker.markTime=42513, ConflictMarker.nodeCount=12, ConflictIdSorter.graphTime=13603, ConflictIdSorter.topsortTime=19241, ConflictIdSorter.conflictIdCount=9, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=219320, ConflictResolver.conflictItemCount=12, DfDependencyCollector.collectTime=29410358, DfDependencyCollector.transformTime=356789}
---
> [DEBUG] Dependency collection stats {ConflictMarker.analyzeTime=42909, ConflictMarker.markTime=19323, ConflictMarker.nodeCount=12, ConflictIdSorter.graphTime=11289, ConflictIdSorter.topsortTime=36168, ConflictIdSorter.conflictIdCount=9, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=201013, ConflictResolver.conflictItemCount=12, DfDependencyCollector.collectTime=26022745, DfDependencyCollector.transformTime=329108}
1039c1017
< [DEBUG]   (f) session = org.apache.maven.execution.MavenSession@3344d163
---
> [DEBUG]   (f) session = org.apache.maven.execution.MavenSession@16f7f59f
1132c1110
< [DEBUG] env.TERM_SESSION_ID: b1021635-e177-4298-8d3a-4d11aa94ad16
---
> [DEBUG] env.TERM_SESSION_ID: a401b2c0-876e-41c0-944c-eae704ec7d6e
1193c1171
< [DEBUG] jsonschema2pojo.version: 1.0.0-alpha2
---
> [DEBUG] jsonschema2pojo.version: 1.0.0-alpha4
1205c1183
< [DEBUG] maven.build.timestamp: 2023-08-12T17:38:06Z
---
> [DEBUG] maven.build.timestamp: 2023-08-12T17:36:43Z
1280c1258
< [DEBUG] Dependency collection stats {ConflictMarker.analyzeTime=50452, ConflictMarker.markTime=39844, ConflictMarker.nodeCount=17, ConflictIdSorter.graphTime=17116, ConflictIdSorter.topsortTime=26132, ConflictIdSorter.conflictIdCount=12, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=288029, ConflictResolver.conflictItemCount=17, DfDependencyCollector.collectTime=44545874, DfDependencyCollector.transformTime=445380}
---
> [DEBUG] Dependency collection stats {ConflictMarker.analyzeTime=42061, ConflictMarker.markTime=40206, ConflictMarker.nodeCount=17, ConflictIdSorter.graphTime=17205, ConflictIdSorter.topsortTime=26015, ConflictIdSorter.conflictIdCount=12, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=241976, ConflictResolver.conflictItemCount=17, DfDependencyCollector.collectTime=45251232, DfDependencyCollector.transformTime=396917}
1332c1310
< [DEBUG]   (f) session = org.apache.maven.execution.MavenSession@3344d163
---
> [DEBUG]   (f) session = org.apache.maven.execution.MavenSession@16f7f59f
1353c1331,1370
< [INFO] Nothing to compile - all classes are up to date
---
> [DEBUG] Stale source detected: /home/tony/path/to/my/module/target/generated-sources/path/to/generated/Class.java
> [INFO] Changes detected - recompiling the module! :source
> [DEBUG] Classpath:
> [DEBUG]  /home/tony/path/to/my/module/target/classes
> [DEBUG]  /home/tony/.m2/repository/org/hibernate/validator/hibernate-validator/6.1.7.Final/hibernate-validator-6.1.7.Final.jar
> [DEBUG]  /home/tony/.m2/repository/jakarta/validation/jakarta.validation-api/2.0.2/jakarta.validation-api-2.0.2.jar
> [DEBUG]  /home/tony/.m2/repository/org/jboss/logging/jboss-logging/3.4.3.Final/jboss-logging-3.4.3.Final.jar
> [DEBUG]  /home/tony/.m2/repository/com/fasterxml/classmate/1.5.1/classmate-1.5.1.jar
> [DEBUG]  /home/tony/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.13.3/jackson-annotations-2.13.3.jar
> [DEBUG]  /home/tony/.m2/repository/org/apache/commons/commons-lang3/3.6/commons-lang3-3.6.jar
> [DEBUG]  /home/tony/.m2/repository/org/springframework/hateoas/spring-hateoas/1.5.5/spring-hateoas-1.5.5.jar
> [DEBUG]  /home/tony/.m2/repository/org/springframework/spring-aop/5.3.29/spring-aop-5.3.29.jar
> [DEBUG]  /home/tony/.m2/repository/org/springframework/spring-beans/5.3.29/spring-beans-5.3.29.jar
> [DEBUG]  /home/tony/.m2/repository/org/springframework/spring-context/5.3.29/spring-context-5.3.29.jar
> [DEBUG]  /home/tony/.m2/repository/org/springframework/spring-expression/5.3.29/spring-expression-5.3.29.jar
> [DEBUG]  /home/tony/.m2/repository/org/springframework/spring-core/5.3.29/spring-core-5.3.29.jar
> [DEBUG]  /home/tony/.m2/repository/org/springframework/spring-jcl/5.3.29/spring-jcl-5.3.29.jar
> [DEBUG]  /home/tony/.m2/repository/org/springframework/spring-web/5.3.29/spring-web-5.3.29.jar
> [DEBUG]  /home/tony/.m2/repository/org/springframework/plugin/spring-plugin-core/2.0.0.RELEASE/spring-plugin-core-2.0.0.RELEASE.jar
> [DEBUG]  /home/tony/.m2/repository/com/jayway/jsonpath/json-path/2.7.0/json-path-2.7.0.jar
> [DEBUG]  /home/tony/.m2/repository/net/minidev/json-smart/2.4.11/json-smart-2.4.11.jar
> [DEBUG]  /home/tony/.m2/repository/net/minidev/accessors-smart/2.4.11/accessors-smart-2.4.11.jar
> [DEBUG]  /home/tony/.m2/repository/org/ow2/asm/asm/9.3/asm-9.3.jar
> [DEBUG]  /home/tony/.m2/repository/javax/annotation/javax.annotation-api/1.3.2/javax.annotation-api-1.3.2.jar
> [DEBUG]  /home/tony/.m2/repository/ch/qos/logback/logback-core/1.3.8/logback-core-1.3.8.jar
> [DEBUG]  /home/tony/.m2/repository/ch/qos/logback/logback-classic/1.3.8/logback-classic-1.3.8.jar
> [DEBUG]  /home/tony/.m2/repository/org/slf4j/slf4j-api/2.0.7/slf4j-api-2.0.7.jar
> [DEBUG]  /home/tony/.m2/repository/org/slf4j/jul-to-slf4j/1.7.36/jul-to-slf4j-1.7.36.jar
> [DEBUG] Source roots:
> [DEBUG]  /home/tony/path/to/my/module/src/main/java
> [DEBUG]  /home/tony/path/to/my/module/target/generated-sources
> [DEBUG]  /home/tony/path/to/my/module/target/generated-sources/annotations
> [DEBUG] Command line options:
> [DEBUG] -d /home/tony/path/to/my/module/target/classes -classpath /home/tony/path/to/my/module/target/classes:/home/tony/.m2/repository/org/hibernate/validator/hibernate-validator/6.1.7.Final/hibernate-validator-6.1.7.Final.jar:/home/tony/.m2/repository/jakarta/validation/jakarta.validation-api/2.0.2/jakarta.validation-api-2.0.2.jar:/home/tony/.m2/repository/org/jboss/logging/jboss-logging/3.4.3.Final/jboss-logging-3.4.3.Final.jar:/home/tony/.m2/repository/com/fasterxml/classmate/1.5.1/classmate-1.5.1.jar:/home/tony/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.13.3/jackson-annotations-2.13.3.jar:/home/tony/.m2/repository/org/apache/commons/commons-lang3/3.6/commons-lang3-3.6.jar:/home/tony/.m2/repository/org/springframework/hateoas/spring-hateoas/1.5.5/spring-hateoas-1.5.5.jar:/home/tony/.m2/repository/org/springframework/spring-aop/5.3.29/spring-aop-5.3.29.jar:/home/tony/.m2/repository/org/springframework/spring-beans/5.3.29/spring-beans-5.3.29.jar:/home/tony/.m2/repository/org/springframework/spring-context/5.3.29/spring-context-5.3.29.jar:/home/tony/.m2/repository/org/springframework/spring-expression/5.3.29/spring-expression-5.3.29.jar:/home/tony/.m2/repository/org/springframework/spring-core/5.3.29/spring-core-5.3.29.jar:/home/tony/.m2/repository/org/springframework/spring-jcl/5.3.29/spring-jcl-5.3.29.jar:/home/tony/.m2/repository/org/springframework/spring-web/5.3.29/spring-web-5.3.29.jar:/home/tony/.m2/repository/org/springframework/plugin/spring-plugin-core/2.0.0.RELEASE/spring-plugin-core-2.0.0.RELEASE.jar:/home/tony/.m2/repository/com/jayway/jsonpath/json-path/2.7.0/json-path-2.7.0.jar:/home/tony/.m2/repository/net/minidev/json-smart/2.4.11/json-smart-2.4.11.jar:/home/tony/.m2/repository/net/minidev/accessors-smart/2.4.11/accessors-smart-2.4.11.jar:/home/tony/.m2/repository/org/ow2/asm/asm/9.3/asm-9.3.jar:/home/tony/.m2/repository/javax/annotation/javax.annotation-api/1.3.2/javax.annotation-api-1.3.2.jar:/home/tony/.m2/repository/ch/qos/logback/logback-core/1.3.8/logback-core-1.3.8.jar:/home/tony/.m2/repository/ch/qos/logback/logback-classic/1.3.8/logback-classic-1.3.8.jar:/home/tony/.m2/repository/org/slf4j/slf4j-api/2.0.7/slf4j-api-2.0.7.jar:/home/tony/.m2/repository/org/slf4j/jul-to-slf4j/1.7.36/jul-to-slf4j-1.7.36.jar: -sourcepath /home/tony/path/to/my/module/src/main/java:/home/tony/path/to/my/module/target/generated-sources:/home/tony/path/to/my/module/target/generated-sources/annotations: -s /home/tony/path/to/my/module/target/generated-sources/annotations -g -target 17 -source 1.8 -encoding UTF-8
> [DEBUG] incrementalBuildHelper#beforeRebuildExecution
> [INFO] Compiling 2 source files with javac [debug target 17] to target/classes
> [DEBUG] incrementalBuildHelper#afterRebuildExecution
> [WARNING] bootstrap class path not set in conjunction with -source 8
> [WARNING] unknown enum constant javax.annotation.meta.When.MAYBE
>   reason: class file for javax.annotation.meta.When not found
1357,1358c1374,1375
< [INFO] Total time:  2.154 s
< [INFO] Finished at: 2023-08-12T19:38:06+02:00
---
> [INFO] Total time:  2.777 s
> [INFO] Finished at: 2023-08-12T19:36:44+02:00
1363c1380
< [WARNING]  * org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha2
---
> [WARNING]  * org.jsonschema2pojo:jsonschema2pojo-maven-plugin:1.0.0-alpha4

```

But I can just see the symptom there, the file is detected as stale for some reason and recompilation is done.

I tried narrowing it down in the code but did not have enough time for that. I am going on a vacation now so I will not respond for the next week but after that I can provide additional details if needed.

**Comment by Indy1138** [comment]
Hello, I would like to add to this issue.

I am experiencing the same behavior, the plugin regenerates the sources each time it is invoked (as expected),
and then maven compiler plugin detects the files as stale and recompiles.
I'm using jsonschema2pojo-maven-plugin 1.2.1, maven-compiler-plugin 3.12.1 with maven 3.9.3

Would it be possible to implement a check such as michakurt proposed?

The protobuf-maven-plugin has for example the checkStaleness parameter.
https://www.xolstice.org/protobuf-maven-plugin/compile-mojo.html
"Normally protoc is invoked on every execution of the plugin.
Setting this parameter to true will enable checking timestamps of source protobuf definitions vs. generated sources."

Something like this would be very helpful for this plugin as well.
